test/xpu_api/utilities/optional/optional.object/optional.object.observe/op_arrow_const.pass.cpp

```diff
@@ -0,0 +1,95 @@
+// -*- C++ -*-
+//===----------------------------------------------------------------------===//
+//
+// Copyright (C) Intel Corporation
+//
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+// This file incorporates work covered by the following copyright and permission
+// notice:
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+//
+//===----------------------------------------------------------------------===//
+
+#include "support/test_config.h"
+
+#include <oneapi/dpl/optional>
+#include <oneapi/dpl/type_traits>
+
+#include "support/test_macros.h"
+#include "support/utils.h"
+
+#if TEST_DPCPP_BACKEND_PRESENT
+using dpl::optional;
+
+struct X
+{
+    constexpr int
+    test() const
+    {
+        return 3;
+    }
```

Comment:
I guess we also need a non-const overload returning different value.
I propose to merge tests for const and non-const overloads of `operator->` into one test as proposed for `operator*`.

Reply:
Fixed